### PR TITLE
Added value class support

### PR DESCRIPTION
**Pull request by LarsSven**
This MR serves to add support for value classes to Mockito Kotlin. Value classes currently break and there is no easy way around them, many users have had issues with this: https://github.com/mockito/mockito-kotlin/issues/445

This MR adds an anyValueClass() and adds it to the any() and anyOrNull() implementation. I expect I may have done a few things non-optimally as I am not too familiar with Mockito's inner workings, but I look forward to feedback from someone who knows more about Mockito argument matchers than me to help me improve this. I do think it is important to get support for value classes.

### CODE REVIEW

File: mockito-kotlin/src/main/kotlin/org/mockito/kotlin/Matchers.kt [hunk 1]
```diff
@@ -29,6 +29,7 @@ import org.mockito.ArgumentMatcher
 import org.mockito.ArgumentMatchers
 import org.mockito.kotlin.internal.createInstance
 import kotlin.reflect.KClass
+import kotlin.reflect.jvm.internal.impl.load.kotlin.JvmType.Object
```
Comment (TimvdLippe) [inline mockito-kotlin/src/main/kotlin/org/mockito/kotlin/Matchers.kt:32]: I don't see a usage of this class in this file. Is this import necessary and the additional dependency?

Comment (LarsSven) [reply mockito-kotlin/src/main/kotlin/org/mockito/kotlin/Matchers.kt]: You were right that the import was unnecessary. I removed it.

The dependency unfortunately still is. The dependency enables reflection for Kotlin, which is necessary for detecting and introspecting value classes.

**Comment by TimvdLippe** [comment]
Not sure why, but tests are failing on this PR. Can you please take a look and ensure all tests are passing?

**Comment by LarsSven** [comment]
I resolved the issues with the tests, it was a relatively obscure issue that had to do with verify not being called for the test where we expected it to error out. I also went ahead and added more tests to test other scenarios like around null vs non-null and nested value classes.

**Comment by vans239** [comment]
Is there any plan to add support for `eq` function?

**Comment by vans239** [comment]
Maybe something like this?
```
inline fun <reified T: Any> eqValue(v: T): T {
    if (!T::class.isValue) return eq(v)
    val field = T::class.java.declaredFields.first { !Modifier.isStatic(it.modifiers) }.apply { isAccessible = true }
    val method = T::class.java.getDeclaredMethod("box-impl", field.type).apply { isAccessible = true }
    argThat(object : ArgumentMatcher<T> {
        override fun matches(argument: T) = v == (method.invoke(null, argument) as T)
        override fun toString() = ValuePrinter.print(v)
    })
    return method.invoke(null, Primitives.defaultValue(field.get(v)::class.java)) as T
}
```
